controllers/pattern_controller.go

```diff
@@ -480,11 +468,63 @@ func (r *PatternReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return err
 	}
 	r.driftWatcher, _ = newDriftWatcher(r.Client, mgr.GetLogger(), newGitClient())
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&api.Pattern{}).
 		Complete(r)
 }
 
+func (r *PatternReconciler) deployGitopsSubscription(filename string) error {
+
+	storedSubscription, _ := r.olmClient.getSubscription(gitopsSubscriptionName, subscriptionNamespace)
+	var newSpec *operatorv1alpha1.Subscription
+
+	// Check if the config map exists and read the config map values
+	if cm, err := r.fullClient.CoreV1().ConfigMaps(common.OperatorNamespace).Get(context.Background(), common.OperatorConfigFile, metav1.GetOptions{}); err == nil {
+		// Config Map exists
```

Comment:
Generally speaking, check for err != nil and return.  That way you avoid needing an else block.

```suggestion
        var cm ConfigMap
	if cm, err := r.fullClient.CoreV1().ConfigMaps(common.OperatorNamespace).Get(context.Background(), common.OperatorConfigFile, metav1.GetOptions{}); err != nil && !kerrors.IsNotFound(err) {
 		return err
 	}
```

Reply:
Updated.